src/plugins/smith_waterman/src/SmithWatermanAlgorithm.cpp

```diff
@@ -173,7 +173,7 @@ void SmithWatermanAlgorithm::calculateMatrixForMultipleAlignmentResult() {
 
     n = pat_n * 2;
     int dirn = (4 + pat_n + 3) >> 2;
-    int *buf, *matrix = (int *)malloc(n * sizeof(int) + pat_n * 0x80 + matrixLength * dirn);
+    int *buf, *matrix = new int[n * sizeof(int) + pat_n * 0x80 + matrixLength * dirn];
     char *score, *score1 = (char *)(matrix + n);
     unsigned char *dir, *dir2, *dir1 = (unsigned char *)score1 + pat_n * 0x80;
     memset(matrix, 0, n * sizeof(int));
```

Comment:
While this is not directly related to the fix it it looks strange to call malloc () memset(0) when you can use a single calloc() instead.
Leaving it up to you if to fix it or not.